File: Soe.Common/DTO/ImportSelectionGridRowDTO.cs
using SoftOne.Soe.Common.Attributes;
namespace SoftOne.Soe.Common.DTO
{
	[TSInclude]
	public class ImportSelectionGridRowDTO
	{
		public string FileName { get; set; }
		public string FileType { get; set; }
		public int DataStorageId { get; set; }
		public ImportDTO Import { get; set; }
		public int? ImportId { get; set; }
		public string ImportName { get; set; }
		public string Message { get; set; }

		public ImportSelectionGridRowDTO(string fileName, string fileType, int dataStorageId, ImportDTO import, string message = "")
		{
			FileName = fileName;
			FileType = fileType;
			DataStorageId = dataStorageId;
			Import = import;
			ImportId = import?.ImportId;
			ImportName = import?.Name;
			Message = message;
		}
	}
}

File: Soe.Angular/TypeScript/Common/Models/ImportSelectionGridRowDTO.ts
import { IImportDTO, IImportSelectionGridRowDTO } from "../../Scripts/TypeLite.Net4";

export class ImportSelectionGridRowDTO implements IImportSelectionGridRowDTO {
    public fileName: string;
    public fileType: string;
    public dataStorageId: number;
    public import: IImportDTO;
    public importId?: number;
    public importName?: string;
    public message: string;
    public doImport: boolean;
    public disableImport: boolean;

    constructor(
        FileName: string,
        FileType: string,
        DataStorageId: number,
        Import: IImportDTO,
        Message: string = "",
        DoImport: boolean | undefined
    ) {
        this.fileName = FileName;
        this.fileType = FileType;
        this.dataStorageId = DataStorageId;
        this.import = Import;
        this.importId = Import?.importId;
        this.importName = Import?.name;
        this.message = Message;
        this.doImport = DoImport ?? Message === "";
        this.disableImport = Import == undefined;
    }
}